src/Features/Core/Portable/Completion/CompletionService.ProviderManager.cs

```diff
@@ -74,6 +74,9 @@ private ImmutableArray<CompletionProvider> GetImportedAndBuiltInProviders(Immuta
                     {
                         providers = GetImportedAndBuiltInProvidersWorker(roles);
                         _rolesToProviders.Add(roles, providers);
+
+                        foreach (var provider in providers)
+                            _nameToProvider[provider.Name] = provider;
```

Comment:
should this use add, or are duplicates OK to be overwritten?

Reply:
It's OK to overwrite, as we do not expect conflict in provider names